### PR TITLE
New testing framework

### PR DESCRIPTION
**Pull request by qnikst**

### CODE REVIEW

File: testing/thor/LICENSE-APACHE [hunk 1]
```diff
@@ -0,0 +1,201 @@
+                              Apache License
```
Comment (mzabaluev) [inline testing/thor/LICENSE-APACHE]: Hmm, this subfolder now claims to have an Apache/MIT license, but jormungandr as a whole has no declared license. We should give some care about this.

File: testing/thor/index.js [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+}
+
+window.document.getElementById("sign_btn").onclick = function(self) {
+   console.log("Creating a builder.");
+   var txbuilder = mjolnir.TransactionBuilder.new();
+   console.log("Adding inputs:");
+   inputs.forEach(function (input, i) {
+      console.log("  input: ", input);
+      var tx = mjolnir.TransactionId.from_hex(input.in_txid);
+      var utxo = mjolnir.UtxoPointer.new(tx, input.in_idx, BigInt(input.out_value));
```
Comment (mzabaluev) [inline testing/thor/index.js]: This is not idiomatic JS, can we bind to the JS class constructor?

File: testing/thor/index.js [hunk 3]
```diff
@@ -0,0 +1,155 @@
...
+        console.log(output);
+        var li = document.createElement("li");
+        var a = document.createElement("a")
+        var t = document.createTextNode("" + output.value + " " + output.address);
+        total += parseInt(output.value);
+        li.appendChild(a);
+        a.appendChild(t);
+        a.setAttribute("class","clickable");
+        a.onclick = function() { alert('hi!') };
+        output_block.appendChild(li);
```
Comment (mzabaluev) [inline testing/thor/index.js]: A less tedious, more maintainable way to go about it is to have a markup template for a list item somewhere in the HTML document, e.g. in an invisible `<div/>`. When a new item is created, clone it into a document fragment and populate the fragment with values, then append the fragment as a child to the list.

File: testing/thor/LICENSE-APACHE [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+   To apply the Apache License to your work, attach the following
+   boilerplate notice, with the fields enclosed by brackets "[]"
+   replaced with your own identifying information. (Don't include
+   the brackets!)  The text should be enclosed in the appropriate
+   comment syntax for the file format. We also recommend that a
+   file or class name and description of purpose be included on the
+   same "printed page" as the copyright notice for easier
+   identification within third-party archives.
+
+Copyright [yyyy] [name of copyright owner]
```
Comment (NicolasDP) [inline testing/thor/LICENSE-APACHE]:
```suggestion
Copyright 2019 Input Output HK Ltd
```

File: testing/thor/index.js (hunk 2 above)
Comment (qnikst) [reply testing/thor/index.js]: not sure if I can do that easily, the value is not a normal JS clas.

**Comment by qnikst** [comment]
I've updated PR.

File: testing/thor/index.js (hunk 3 above)
Comment (qnikst) [reply testing/thor/index.js]: Here were are clearing multiple `<li>` so I expect that amount of code and complexities would be pretty much similar, but thanks, for the idea, I'll use that if representation will be more complex.

**Pull request merged**